### PR TITLE
connect fsm with handshake

### PR DESCRIPTION
**Pull request by chienwei-lan**
check connection with your peer while load driver

close connection while unload driver

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

**Comment by chienwei-lan** [comment]
sqlite3.OperationalError: database is locked

File: src/runtime_src/driver/xclng/drm/xocl/subdev/mailbox.c [hunk 1]
```diff
@@ -1368,7 +1512,22 @@ static void process_request(struct mailbox *mbx, struct mailbox_msg *msg)
 		}
 	} else if (req->req == MAILBOX_REQ_TEST_READY) {
 		MBX_INFO(mbx, "%s: %d", recvstr, req->req);
-	} else if (mbx->mbx_listen_cb) {
+	} else if (req->req == MAILBOX_REQ_CONN_EXPL) {
+		MBX_INFO(mbx, "%s: %d", recvstr, req->req);
+		if(mbx->mbx_state != CONN_SYN_SENT){
```
Comment (maxzhen) [inline src/runtime_src/driver/xclng/drm/xocl/subdev/mailbox.c]: Shouldn't the state handled internally by your state machine handler?

**Comment by maxzhen** [comment]
retest this please

**Pull request merged**